cluster-autoscaler/simulator/predicatechecker/schedulerbased.go

```diff
@@ -126,15 +126,8 @@ func (p *SchedulerBasedPredicateChecker) FitsAnyNodeMatching(clusterSnapshot clu
 			continue
 		}
 
-		filterStatuses := p.framework.RunFilterPlugins(context.TODO(), state, pod, nodeInfo)
-		ok := true
-		for _, filterStatus := range filterStatuses {
-			if !filterStatus.IsSuccess() {
-				ok = false
-				break
-			}
-		}
-		if ok {
+		filterStatus := p.framework.RunFilterPlugins(context.TODO(), state, pod, nodeInfo)
+		if filterStatus == nil || filterStatus.IsSuccess() {
```

Comment:
nit: Status.IsSuccess already returns true for nil statuses, so I don't think the nil check is needed.

Reply:
It's actually the other way around.
Current implementation returns nil for success. Adding IsSuccess() is "future-proof" incase they return non-nil obj for success=true

Reply:
As discussed offline, `IsSuccess` already handled status being nil, so we don't have to do it again.

Reply:
offline conversation to remove  nil check. because nil obj is handled by `IsSuccess` func